CHANGES.md

```diff
@@ -1,6 +1,10 @@
 ## ?
+### Added
+- Add `Conformist.list` to support decoding list of `'a`
+
 ### Changed
 - `Conformist.Field.optional` is now `Conformist.Field.is_optional` to avoid confusion with `Conformist.optional`
+- The encoded value is assumed to be `string list` instead of `string`. This is preparation to support decoding of a list of strings.
```

Comment:
Will the elements be non-lists in the future?

Reply:
This is more of an internal change actually. The input is always `(string * string list)` but we just took the head of the value list and now we are passing the `string list` to decoders and validators.

Reply:
But it does affect the decoder and encoder that the user has to write, right?

Reply:
You are right

Reply:
Would it be possible to adjust that or do you think we can just leave it like that?

Reply:
What do you want to adjust? The decoder takes `string list` as input, because this is what we define as input now. Do you think that should be different?

Reply:
Well that makes the API less clear for users IMO. For example for
```ocaml
let occupation_encoder = function
  | Mathematician -> [ "mathematician" ]
  | Engineer -> [ "engineer" ]
;;
```
- Why do I have to provide a list?
- What happens if I have multiple values in the list?
- What happens if the list is empty?

Reply:
> Why do I have to provide a list?

Are you asking from a perspective of a new user?
Well, if we want to support lists we have to extend the input value from `string` to `string list`. If we want to expose conformists power to the end user, then we have to have the API that you just posted. For ergonomics we could introduce a "simple" encoder/decoder pair, which also clutters the API.

If we don't expose all of conformists power to the end user, it wouldn't be possible for instance to implement custom tuple types nicely.

Throughout the documentation we simply stop talking about key-value semantics and instead talk about key-value*s* semantics.

> What happens if I have multiple values in the list?

That is up to you, it is your type after all :)

> What happens if the list is empty?

Same here

Reply:
Ah ok that makes sense